docroot/profiles/custom/cgov_site/modules/custom/cgov_blog/cgov_blog.tokens.inc

```diff
@@ -139,10 +142,47 @@ function _get_blog_topic_path($term) {
  * Get the blog series title base of the url params..
  */
 function _get_blog_series_title($node) {
-  $year = \Drupal::request()->query->get('year');
-  $topic = \Drupal::request()->query->get('topic');
-  $month = \Drupal::request()->query->get('month');
 
-  $blogManager = \Drupal::service('cgov_blog.blog_manager');
-  return $blogManager->getBlogSeriesTitle($month, $year, $topic, $node);
+  // Get year parameter.
+  $year_param = \Drupal::request()->query->get('year');
+  // Filters HTML to prevent cross-site-scripting (XSS) vulnerabilities.
+  $year = Xss::filter($year_param);
+  // Escapes text by converting special characters to HTML entities.
+  $year = Html::escape($year);
+
+  // Get topic parameter.
+  $topic_param = \Drupal::request()->query->get('topic');
+  // Filters HTML to prevent cross-site-scripting (XSS) vulnerabilities.
+  $topic = Xss::filter($topic_param);
+  // Escapes text by converting special characters to HTML entities.
+  $topic = Html::escape($topic);
+
+  // Get month parameter.
+  $month_param = \Drupal::request()->query->get('month');
+  // Filters HTML to prevent cross-site-scripting (XSS) vulnerabilities.
+  $month = Xss::filter($month_param);
+  // Escapes text by converting special characters to HTML entities.
+  $month = Html::escape($month);
+
+  // Return NULL if the year, month and topic parameters are NULL.
+  if ($year == NULL && $month == NULL && $topic == NULL) {
+    return NULL;
+  }
+
+  // Make sure For topic, only alphanumeric characters should be
+  // allowed, plus colon ( : ) and hyphen ( - ).
+  if (!empty($topic) && preg_match('/^[a-zA-Z0-9:-]*$/', $topic)) {
+    $blogManager = \Drupal::service('cgov_blog.blog_manager');
+    return $blogManager->getBlogSeriesTitle($month, $year, $topic, $node);
+  }
+
+  // Make sure For year, only four digit numbers should be allowed.
+  // Make sure For moth, only numbers 01-12 should be allowed.
+  if ((is_numeric($year) && strlen($year) == 4) && (is_numeric($month) && $month > 0 && $month <= 12)) {
+    $blogManager = \Drupal::service('cgov_blog.blog_manager');
+    return $blogManager->getBlogSeriesTitle($month, $year, $topic, $node);
+  }
+
+  // Return 400 Bad Request if any input is invalid.
+  throw new BadRequestHttpException('Bad Request');
```

Comment:
@blairlearn I updated the PR with the changes mentioned above. Please review. Thanks

Reply:
Your changes to `_get_blog_series_title()` are more complicated than they need to be.

For `$year` and `$month`, all you need is this. No need to run them through `Xss::filter` and `Html::escape`.

```php
    if(!is_numeric($year) || $year < 0) {
        throw new throw new BadRequestHttpException('year is invalid');
    }
    if(!is_numeric($month) || $month < 1 || $month > 12) {
        throw new throw new BadRequestHttpException('month is invalid');
    }
```

It's going to be similar for `$topic`. As soon as you know it's invalid, throw `BadRequestHttpException` and be done with it.  No need for extra logic for the next person to try to figure out what it's supposed to do. (That "next person" might be you six months from now, trying to figure out what this code does.)

A few more things on `$topic`:

The only characters you're allowing are a-z, A-Z, 0-9, colon, and hyphen.  I don't see any discussion in the comments on #3402, so I'm not sure where that list of characters came from.  (They're awfully close to what I suggested in [my first comment on the ticket](https://github.com/NCIOCPL/cgov-digital-platform/issues/3402#issuecomment-1155450637); I hope that wasn't the sole source of information.)

I like that you're using `Xss::filter` and `Html::escape` to sanitize the inputs rather than trying to "roll your own".  That being said, have you considered the inputs and outputs of those functions? For `$month` and `$year`, the only valid inputs (numbers) wouldn't actually change.  (That is `Xss::filter('1')` is going to be `'1'`.  Same thing for `Html::escape`.) And since you're then checking for whether you got numbers....  There really wasn't any point in calling those functions.  (That being said, it's best to err on the side of safety.)

One final thing, your comments seem to be documenting exactly what `Xss::filter` and `Html::escape` do. I could Google the Drupal API to get that information. A more useful comment would be one, written in your own words, explaining what the code is _intended_ to do.  (e.g. "Sanitize query string inputs to prevent string injection.")

Reply:
@blairlearn Thanks for your review.

I've made the changes as per your comments. Please review the latest changes.

@jfrank-nih Can you please confirm the below topic validation.
**only alphanumeric characters should be allowed, plus colon ( : ) and hyphen ( - )**